va/caa.go

```diff
@@ -81,15 +83,15 @@ func (va *ValidationAuthorityImpl) IsCAAValid(ctx context.Context, req *vapb.IsC
 					req.Domain,
 					req.AccountURIID,
 					string(validationMethod),
-					prob,
+					nil,
```

Comment:
@pgporada I wanted to particularly get your eyes on this line and the one below. By my reading of the conditionals here, `prob` was always `nil` at these points in the code. So I just replaced with the literal `nil`.

Since these are the only call sites, if that's right, we can also simplify processRemoteCAAResults.

Reply:
This was an oversight on my part when I copied this chunk of code from the `//va/va.go` and failed to do enough Println investigation. Running this `./n.sh -u -p ./va -f TestMultiCAARechecking -v` with some debugging lines inside `processRemoteCAAResults` shows only `nil` for the `primaryResults *probs.ProblemDetails` variable.

Reply:
We did some screensharing and determined the two occurrences of `prob` in calls to `va.processRemoteValidationResults` can be changed to `nil`. Instead of that though, we should just remove `primaryResult *probs.ProblemDetails` from that method signature.